lib/ukschedcoop/schedcoop.c

```diff
@@ -264,6 +264,17 @@ static int schedcoop_start(struct uk_sched *s,
 	return 0;
 }
 
+static const struct uk_thread *schedcoop_idle_thread(struct uk_sched *s,
+						     unsigned int proc_id)
+{
+	struct schedcoop *c = uksched2schedcoop(s);
+
+	/* NOTE: We only support one processing LCPU for now */
+	if (proc_id > 0)
+		return NULL;
```

Comment:
Nitpick. Probably just my personal taste, but I think it would look nicer with a newline before the final `return` statement :thinking:. Otherwise I like having it as `return proc_id > 0 ? NULL : &(c->idle);`, if the final return statement depends on this parameter, but this looks more like just an early check for the function argument to me :thinking:.

Reply:
I put a newline after the first `return` that is fine, otherwise this function is intended as somewhat a skeleton for a more complete implementation, meaning `schedcoop` supporting multiple LCPUs. Because of this I prefer the function structure as is.